rasa/core/lock_store.py

```diff
@@ -305,7 +311,17 @@ def _create_from_endpoint_config(
 
         lock_store: LockStore = InMemoryLockStore()
     elif endpoint_config.type == "redis":
+        rasa.shared.utils.io.raise_deprecation_warning(
+            "'RedisLockStore' is deprecated and will be removed in Rasa Open "
+            "Source 4.0.0. Please migrate to the enhanced 'ConcurrentRedisLockStore'.",
+            docs=DOCS_URL_MIGRATION_GUIDE,
+        )
```

Comment:
Are the docs going to be handled separately or still incoming for this pr?

Reply:
I think docs should be added separately once the implementation goes through review and then gets merged. I will create a separate task for this.

Reply:
Added here: https://rasahq.atlassian.net/browse/ATO-211